_studio/mfx_lib/scheduler/src/mfx_scheduler_core_ischeduler.cpp

```diff
@@ -68,27 +68,15 @@ mfxStatus mfxSchedulerCore::Initialize2(const MFX_SCHEDULER_PARAM2 *pParam)
     }
 
     // clean up the task look up table
-    umcRes = m_ppTaskLookUpTable.Alloc(MFX_MAX_NUMBER_TASK);
-    if (UMC::UMC_OK != umcRes)
-    {
-        return MFX_ERR_MEMORY_ALLOC;
-    }
+    m_ppTaskLookUpTable.resize(MFX_MAX_NUMBER_TASK, nullptr);
 
     // allocate the dependency table
-    m_pDependencyTable.Alloc(MFX_MAX_NUMBER_TASK * 2);
-    if (UMC::UMC_OK != umcRes)
-    {
-        return MFX_ERR_MEMORY_ALLOC;
-    }
+    m_pDependencyTable.resize(MFX_MAX_NUMBER_TASK * 2, MFX_DEPENDENCY_ITEM());
```

Comment:
As far as I see it is the only place where m_pDependencyTable can be changed. If so then no need to check in many places for (smth < MFX_MAX_NUMBER_TASK * 2). If not so, then check only where m_pDependencyTable.size() grows.